### PR TITLE
Docs rehaul for v0.8

### PR DESCRIPTION
**Pull request by theabhirath**
This is the last PR for v0.8, and introduces just about enough documentation to ensure that users can find their way around the major parts of the repo. A large swathe of the builders remain undocumented, but that will take time and effort and I would rather get v0.8 out first and then worry about that.

cc @darsnack for suggestions in case something important has been missed.

### CODE REVIEW

**Comment by CarloLucibello** [comment]
We should disallow loading densenet pretrained weights and mark the weight as not available in the docs.

File: docs/src/howto/resnet.md [hunk 1]
```diff
@@ -19,3 +19,41 @@ model = ResNet(50; pretrain=true)
...
+base width:
+
+```julia
+custom_resnet = Metalhead.resnet(Metalhead.bottleneck, [3, 4, 6, 3];
+                                 cardinality = 32, base_width = 4,
+                                 pool_layer = Layers.AdaptiveMeanMaxPool((1, 1)),
+                                 stochastic_depth_prob = 0.2)
+```
+
+And we have a custom model, built with minimal effort! The documentation for `Metalhead.resnet` has been written with extensive care and in as much detail as possible to facilitate user ease. However, if you find anything difficult to understand, feel free to open an issue and we will be happy to help you out, and to improve the documentation where necessary.
```
Comment (darsnack) [inline docs/src/howto/resnet.md:59]:
```suggestion
And we have a custom model, built with minimal effort! The documentation for `Metalhead.resnet` has been written with extensive care and in as much detail as possible to facilitate ease of use. Still, if you find anything difficult to understand, feel free to open an issue and we will be happy to help you out, and to improve the documentation where necessary.
```

File: docs/src/api/utilities.md [hunk 2]
```diff
@@ -1,10 +1,8 @@
-# Utilities
+# Model utilities
```
Comment (darsnack) [inline docs/src/api/utilities.md:1]: Rename this in `docs/make.jl` too

**Review by darsnack — CHANGES_REQUESTED** [review]
Looks pretty good to me!

Other details that I cannot directly comment on:
- The `DenseNet` docstring has the "Arguments" bullet list indented which is being interpreted as a markdown code block
- The `UNet` docstring has the same issue as `DenseNet`
- The `MLPMixer` + `ResMLP` docstring should list the config types as symbols (i.e. `:small`, `:base`, etc.)

**Comment by darsnack** [comment]
Also need to remove `DenseNet` from the pretrained models list in test.

**Comment by theabhirath** [comment]
Done, done, done and done! Hopefully tests pass, and then we can finally release this 😁. I have a ready blog post for the Flux website detailing the major improvements as well, so I can land that PR after we tag 😊.

**Comment by theabhirath** [comment]
Can I go ahead and tag v0.8 now?

**Comment by darsnack** [comment]
Yeah go for it

**Pull request merged**